### ISSUE TITLE
Getting 502 Error from Edge Function

### ISSUE THREAD

**Issue by bgandy94** [supabase/supabase#6647]
# Bug report

## Describe the bug
Hey all! I'm building a relatively simple edge function that just takes a request body, validates it, and saves it in the db. and for some reason, I can't figure out why but I keep getting a 502 response from this endpoint.

I've tried deleting and redeploying the function. I've tried commenting out most/all of the function but a response in an effort of trying to isolate if it's a package issue, supabase connection issue, etc.. and nothing has worked.

Here's a codepen with the code for my edge function: https://codesandbox.io/s/strange-wilson-ylnfpp?file=/src/functions/create-event/index.ts

I copied and pasted the example found here: https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts
and that function is working fine. But for some reason my function is returning the following error:

```
502: Bad Gateway (ORIGIN_BOOT_RPC_ERROR)
The subhoster origin returned a failure status for the boot RPC call. Status: 500 Internal Server Error
```

## To Reproduce
* Get the code for my edge function in question here: https://codesandbox.io/s/strange-wilson-ylnfpp?file=/src/functions/create-event/index.ts
* Deploy the function to a supabase project (I'm sure the project won't have an `events` table, that's fine. IF, an error is returned stating that relating doesn't exist, then I'd be ecstatic.
* Hit the function with with your favorite api tool.

## Expected behavior

In my project, I'd expect the function to take the in the body, validate it, store a new record, and return a non 500 response

## System information

- WSL (Ubuntu)
- Version of supabase-js: 1.35.2
- Version of Node.js: 16.13.1

**Comment by inian** [comment]
Hi, can you send the name of your function, project ref in a support ticket [here](https://app.supabase.io/support/new) Thanks!

**Comment by MarcDerhammer** [comment]
We might be experiencing the same sort of problem.  Does yours behave normally when served locally?

https://github.com/supabase/supabase/discussions/6710

**Comment by bgandy94** [comment]
Yes it does @MarcDerhammer

**Comment by MarcDerhammer** [comment]
There might be a memory limit we're reaching.  Maybe there's documentation somewhere I missed, or maybe the docs need to be updated with this info.  Not 100% sure :(

**Comment by bgandy94** [comment]
I would be shocked if this is the case for me, as I started experiencing this behavior on a POC type project that was only called a handful of times prior to hitting this roadblock

**Comment by paddyfink** [comment]
Hello @bgandy94, I'm facing the same issue. Was it because of the script size?

**Comment by bgandy94** [comment]
@paddyfink yes, the script size was the issue! Going to close the issue now, but always curious how we can improve the docs/experience to make this issue more obvious to avoid future issues of the same nature.

**Comment by paddyfink** [comment]
@bgandy94 Is there a way to view the script size in the dashboard? I'm facing the same issue again with another function and still not sure if it is related to the size

**Comment by Silur** [comment]
a note on this limitation would indeed be useful on the pricing page as `ORIGIN_BOOT_RPC_ERROR` does not really imply that it's a deliberate limitation and not an actuall error

**Comment by inian** [comment]
Hi @Silur, you should get an error when deploying a function greater than the script size allowed by your plan. Are not able to see that error during the deployment?

**Comment by JanLoebel** [comment]
I have the same problem, deploying the function works without any error. If I check the bundle size with `deno bundle index.ts > out.bundle` it's about 500kb. So the size should not be the problem even if I'm using the free tier. 
If I remove the dependency: `import { createReport } from 'https://unpkg.com/docx-templates@4.9.2/lib/browser.js';` it works. But with the dependency the OPTIONS call directly fail not even the POST call. The library itself says that for DENO we have to use this import url. Any ideas?

**Comment by inian** [comment]
Can you send a [support ticket](https://app.supabase.com/support/new) @JanLoebel so that we can debug further? You should also be able to see logs in the function tab of your dashboard on what could be causing the 502.

**Comment by JanLoebel** [comment]
@inian Have done that yet, the logs are empty because it already fails before the function starts when the dependency is added.

**Comment by laktek** [comment]
@JanLoebel We recently made a fix that might solve this issue. Can you try deploying your function again?

**Comment by JanLoebel** [comment]
@laktek Thank you for the information. I've redeployed the function and it works! Great, thank you so much!

**Comment by bgandy94** [comment]
I'm not aware of a way to do that. I just knew it was my issue because joi
is a big library
